test/integration/test_keyprotect.py

```diff
@@ -0,0 +1,149 @@
+        for key in self.kp.keys():
+            self.kp.delete(key.get('id'))
+
+        self.rc.delete_instance(self.instance_id)
+        self.kp.session.close()
+
+    def test_wrap_unwrap_rotate_rewrap(self):
+        # wrapping message
+        message = b'This is a really important message.'
+        wrapped = self.kp.wrap(self.key.get('id'), message)
+        ciphertext = wrapped.get("ciphertext")
+        # unwrapping
+        unwrapped = self.kp.unwrap(self.key.get('id'), ciphertext)
+        self.assertEqual(message, unwrapped)
+        # rotate key
+        rotated = self.kp.rotate(self.key.get('id'))
+        # rewrapping
+        rewrapped = self.kp.rewrap(self.key.get('id'), ciphertext)
+        self.assertNotEqual(rewrapped['ciphertext'], ciphertext)
+        self.assertNotEqual(rewrapped['rewrappedKeyVersion']['id'], self.key.get('id'))
+
+    def test_wrap_unwrap_rotate_rewrap_with_aad(self):
+        # wrapping message
+        message = b'This is a really important message.'
+        wrapped = self.kp.wrap(self.key.get('id'), message, aad=['python-keyprotect'])
+        ciphertext = wrapped.get("ciphertext")
+        # unwrapping
+        unwrapped = self.kp.unwrap(self.key.get('id'), ciphertext, aad=['python-keyprotect'])
+        self.assertEqual(message, unwrapped)
+        # rotate key
+        self.kp.rotate(self.key.get('id'))
+        # rewrapping
+        rewrapped = self.kp.rewrap(self.key.get('id'), ciphertext, aad=['python-keyprotect'])
+        self.assertNotEqual(rewrapped['ciphertext'], ciphertext)
+        self.assertNotEqual(rewrapped['rewrappedKeyVersion']['id'], self.key.get('id'))
+
+    def test_restore_key(self):
+        self.kp.delete(self.import_key.get('id'))
+        resp = self.kp.get(self.import_key.get('id'))
+        self.assertEqual(resp['state'], 5)
+
+        # restore, must wait 30 sec or more before sending an restore
+        time.sleep(30)
+        self.kp.restore(self.import_key.get('id'), payload=b"Payload for test")
+        resp = self.kp.get(self.import_key.get('id'))
+        self.assertEqual(resp['state'], 1)
+
+    def test_disable_enable_key(self):
+        # disable
+        self.kp.disable_key(self.key.get('id'))
+        resp = self.kp.get(self.key.get('id'))
+        self.assertEqual(resp['state'], 2)
+
+        # enable, must wait 30 sec or more before sending an enable
+        time.sleep(30)
+        self.kp.enable_key(self.key.get('id'))
+        resp = self.kp.get(self.key.get('id'))
+        self.assertEqual(resp['state'], 1)
+
+    def test_import_get_token(self):
+        # create import token
+        resp = self.kp.create_import_token(86400, 50)
+        self.assertEqual(resp['maxAllowedRetrievals'], 50)
+        self.assertEqual(resp['remainingRetrievals'], 50)
+
+        # get import token
+        resp = self.kp.get_import_token()
+        self.assertEqual(resp['maxAllowedRetrievals'], 50)
+        self.assertEqual(resp['remainingRetrievals'], 49)
```

Comment:
Maybe validate that we actually get back some correct JSON structure with content here as well

https://cloud.ibm.com/apidocs/key-protect#getimporttoken

So I would check to see if both payload and nonce are present and not empty